baker/GrapherBaker.tsx

```diff
@@ -109,28 +102,146 @@ export const renderDataPageOrGrapherPage = async (
+    // rather than for each individual data page. We then attach it to the
+    // current datapageGdoc.
+    datapageGdoc.imageMetadata = imageMetadataDictionary
+
+    const datapageGdocContent =
+        parseGdocContentFromAllowedLevelOneHeadings(datapageGdoc)
+
+    // Passing `datapageGdocContent` as a separate prop rather than weaving it
+    // into datapageGdoc to keep the concept of "Gdoc" clearer
+    return renderToHtmlPage(
+        <DataPage
+            grapher={grapher}
+            variableId={id}
+            datapageJson={datapageJson}
+            datapageGdoc={datapageGdoc}
+            datapageGdocContent={datapageGdocContent}
+            baseUrl={BAKED_BASE_URL}
+            baseGrapherUrl={BAKED_GRAPHER_URL}
+            isPreviewing={false}
+        />
+    )
+}
+
+/**
+ *
+ * Similar to renderDataPageOrGrapherPage(), but for admin previews
+ */
+export const renderPreviewDataPageOrGrapherPage = async (
+    grapher: GrapherInterface,
+    publishedExplorersBySlug?: Record<string, ExplorerProgram>
+) => {
+    const variableIds = uniq(grapher.dimensions!.map((d) => d.variableId))
+    // this shows that multi-metric charts are not really supported, and will
+    // render a datapage corresponding to the first variable found.
+    const id = variableIds[0]
+
+    // Get the datapage JSON file from the owid-content git repo that has
+    // been updated and pulled separately by an author.
+    const { datapageJson, parseErrors } = await getDatapageJson(id)
+
+    // When previewing a datapage we want to show all discrepancies with the
+    // expected JSON schema in the browser.
+    if (parseErrors.length > 0)
+        return renderToHtmlPage(
+            <pre>{JSON.stringify(parseErrors, null, 2)}</pre>
+        )
+
+    // Fallback to rendering a regular grapher page whether the datapage JSON
+    // wasn't found or if it is not set to override the current grapher page
+    if (
+        // This could be folded into the showDataPageOnChartIds check below, but
+        // is kept separate to reiterate that that abscence of a datapageJson leads to
+        // rendering a grapher page
+        !datapageJson ||
+        // We only want to render datapages on selected charts, even if the
+        // variable found on the chart has a datapage configuration.
+        !grapher.id ||
+        !datapageJson.showDataPageOnChartIds.includes(grapher.id)
     )
         return renderGrapherPage(grapher)
 
     // Compliment the text-only content from the JSON with rich text from the
     // companion gdoc
     const datapageGdoc = await getDatapageGdoc(
         datapageJson,
-        isPreviewing,
+        true,
         publishedExplorersBySlug
     )
 
+    const datapageGdocContent =
+        parseGdocContentFromAllowedLevelOneHeadings(datapageGdoc)
+
+    // Passing `datapageGdocContent` as a separate prop rather than weaving it
+    // into datapageGdoc to keep the concept of "Gdoc" clearer
```

Comment:
Just confirming, fragments that are using this heading technique are still having their links tracked, right?

Reply:
You're right to flag this, they don't at the moment, but [they should](https://github.com/owid/owid-grapher/pull/2233#issuecomment-1559173499).